### DISCUSSION TITLE
How to optimize the parameters to make the RMSE fitting value close to 0 when predicting multiple output items through multiple input items?

### DISCUSSION THREAD

**Discussion by bodypumper** [optuna/optuna#5445] [Q&A]
The following is the practical process

1：The experimental data for the original concrete composition ratio is 1000 pieces, and the algorithm model used is the random forest regression model.

2：The following code is for creating models, training models, predicting target values, and optimizing Optuna.
![image](https://github.com/optuna/optuna/assets/44149782/83f4e8a1-97d9-4c17-9879-d1d7e337b402)

The fitting index of RMSE is 27.781625571862275.

**Comment by nzw0301** [comment]
What about the training loss? If it's small, it will be overfitting. Otherwise, `n_trials` looks large enough to converges, so I suppose expanding the hyper-parameters of random forest or use different regressor (neural net?) might reduce the loss.

**Reply by bodypumper** [reply]
> What about the training loss? If it's small, it will be overfitting. Otherwise, `n_trials` looks large enough to converges, so I suppose expanding the hyper-parameters of random forest or use different regressor (neural net?) might reduce the loss.

I have tried all the methods you mentioned above, but the RMSE value remains around 18. My dataset has 1030 entries. Is this quantity not enough?

**Reply by nzw0301** [reply]
> What about the training loss?

**Reply by bodypumper** [reply]
I am currently using Optuna for parameter tuning to avoid training losses. Is there any other way to solve my question?

**Reply by nzw0301** [reply]
I think it is not possible to say something without comparing training loss value and 

>  RMSE value remains around 18.

**Reply by nzw0301** [reply]
To be clear, I meant sharing the training loss, not optimising training loss by optuna.

**Reply by bodypumper** [reply]
> To be clear, I meant sharing the training loss, not optimising training loss by optuna.


First, I employed data preprocessing techniques, including standardization.  
Subsequently, I utilized boosting methods, specifically a blend of decision trees and AdaBoostRegressor.  
I am exploring alternative approaches to address the issue of training loss.

![image](https://github.com/optuna/optuna/assets/44149782/cba43bc8-a5fe-4bfe-a6ef-8424734f5a8b)

**Comment by bodypumper** [comment]
Hello Maintainer. 
I used a downloaded dataset A and also have a test dataset B in my hand. I will use these two sets of data separately to train the model and perform prediction and RMSE calculation. But the results are completely different. The R2 of dataset A is 0.99 and RMSE is 0.02, while the R2 of dataset B is 0.97 and RMSE is 10.46. So I guess this model is essentially usable. However, the value content and data processing of the dataset still need more attention. Is there a possibility of this?

**Comment by bodypumper** [comment]
I have found the reason why the RMSE value cannot be reduced to 1. x. Because the calculation of RMSE is sensitive to 0 values, it is necessary to process the 0 values in “y_true” and "y_pred" before the calculation.
